patches/net/minecraft/locale/Language.java.patch

```diff
@@ -1,34 +1,81 @@
          BiConsumer<String, String> biconsumer = builder::put;
-         parseTranslations(biconsumer, "/assets/minecraft/lang/en_us.json");
+-        parseTranslations(biconsumer, "/assets/minecraft/lang/en_us.json");
 -        final Map<String, String> map = builder.build();
++        Builder<String, net.minecraft.network.chat.Component> componentBuilder = ImmutableMap.builder();
++        parseTranslations(biconsumer, componentBuilder::put, "/assets/minecraft/lang/en_us.json");
 +        final Map<String, String> map = new java.util.HashMap<>(builder.build());
++        Map<String, net.minecraft.network.chat.Component> componentMap = componentBuilder.build();
 +        net.neoforged.neoforge.server.LanguageHook.captureLanguageMap(map);
          return new Language() {
              @Override
              public String getOrDefault(String p_128127_, String p_265421_) {
-@@ -64,6 +_,11 @@
+@@ -64,21 +_,49 @@
                          )
                          .isPresent();
              }
 +
 +            @Override
 +            public Map<String, String> getLanguageData() {
 +                return map;
++            }
++
++            @Override
++            public @org.jetbrains.annotations.Nullable net.minecraft.network.chat.Component getComponent(String key) {
++                return componentMap.get(key);
 +            }
          };
      }
  
-@@ -91,6 +_,9 @@
++    @Deprecated
+     private static void parseTranslations(BiConsumer<String, String> p_282031_, String p_283638_) {
++        parseTranslations(p_282031_, (key, value) -> {}, p_283638_);
++    }
++
++    private static void parseTranslations(BiConsumer<String, String> p_282031_, BiConsumer<String, net.minecraft.network.chat.Component> componentConsumer, String p_283638_) {
+         try (InputStream inputstream = Language.class.getResourceAsStream(p_283638_)) {
+-            loadFromJson(inputstream, p_282031_);
++            loadFromJson(inputstream, p_282031_, componentConsumer);
+         } catch (JsonParseException | IOException ioexception) {
+             LOGGER.error("Couldn't read strings from {}", p_283638_, ioexception);
+         }
+     }
+ 
+     public static void loadFromJson(InputStream p_128109_, BiConsumer<String, String> p_128110_) {
++        loadFromJson(p_128109_, p_128110_, (key, value) -> {});
++    }
++
++    public static void loadFromJson(InputStream p_128109_, BiConsumer<String, String> p_128110_, BiConsumer<String, net.minecraft.network.chat.Component> componentConsumer) {
+         JsonObject jsonobject = GSON.fromJson(new InputStreamReader(p_128109_, StandardCharsets.UTF_8), JsonObject.class);
+ 
+         for (Entry<String, JsonElement> entry : jsonobject.entrySet()) {
++            if (entry.getValue().isJsonArray()) {
++                var component = net.minecraft.network.chat.ComponentSerialization.CODEC.parse(com.mojang.serialization.JsonOps.INSTANCE, entry.getValue()).getOrThrow(com.google.gson.JsonParseException::new);
```

Comment:
I would like to see a better error message (like the translation key and file it's parsing)

Reply:
True. I'll probably also make this a logged error instead of an exception.

Reply:
Added info about the translation key, the file it's parsing isn't passed in and will be logged anyways